device/sycl/src/finding/find_tracks.hpp

```diff
@@ -26,7 +26,6 @@
 #include "traccc/finding/device/find_tracks.hpp"
 #include "traccc/finding/device/make_barcode_sequence.hpp"
 #include "traccc/finding/device/propagate_to_next_surface.hpp"
-#include "traccc/finding/device/prune_tracks.hpp"
```

Comment:
Please add this:

```cpp
// TODO: Remove when updating to future detray version.
#include "detray/core/concepts.hpp"
#include "traccc/finding/device/propagate_to_next_surface.hpp"
```

To see if it removes the issue.

Reply:
This is weird, because I think the `constrained_step` header has the required includes (which would be `#include "detray/definitions/algebra.hpp"`, because it is an algebra concept). However, the constrained step is itself only included in `propagate_to_next_surface.ipp`. So, could you try including 
```
#include <detray/propagator/constrained_step.hpp>
```
in this file https://github.com/acts-project/traccc/blob/45f1cc8c1e041accb78b261e5181b220bbbdb4f1/core/include/traccc/utils/propagation.hpp#L1 ?